front/app/modules/commercial/insights/services/batchAssignment.ts

```diff
@@ -0,0 +1,49 @@
+import { API_PATH } from 'containers/App/constants';
+import streams from 'utils/streams';
+
+const getInsightsBatchEndpoint = (viewId: string) =>
+  `insights/views/${viewId}/batch/`;
+
+// Batch category assignment.
+//
+// Batch assignment is idempotent. It will not complain if some of the
+// assignments already exist.
+//
+// Batch assignment is transactional. Either it succeeds at assigning all
+// categories to all inputs, or the DB is rolled back to its previous state.
+
+export async function batchAssignCategories(
+  insightsViewId: string,
+  inputs: string[],
+  categories: string[]
+) {
+  const res = await streams.add(
+    `${API_PATH}/${getInsightsBatchEndpoint(insightsViewId)}assign_categories`,
+    {
+      inputs,
+      categories,
```

Comment:
This is not exactly a comment for this PR but when we delete categories, we don't send an array of strings but an object: `{. id: some id, type: category }`. I think we should be consistent and I like the array of id-s better, so I wonder if we should change the other endpoints?